### PR TITLE
Preserve local storage when migrating from webview to Crosswalk

### PR DESCRIPTION
**Pull request by fujunwei**
Copy the data of web storage directory to Crosswalk profile name for
Android Kitkat.
The KK webview and the pre-KK webview use different formate to store data,
but the storage database have implemented the mechanism that it will be upgraded
to V2(BLOB value column) if there was V1 (TEXT value column) database already.

BUG=XWALK-4349

### CODE REVIEW

File: runtime/browser/xwalk_browser_main_parts_android.cc [hunk 1]
```diff
@@ -50,12 +53,9 @@ base::StringPiece PlatformResourceProvider(int key) {
   return base::StringPiece();
 }
 
-void MoveUserDataDirIfNecessary(const base::FilePath& user_data_dir,
+void ImportKitkatDataIfNecessary(const base::FilePath& old_data_dir,
                                 const base::FilePath& profile) {
```
Comment (d0ngjun) [inline runtime/browser/xwalk_browser_main_parts_android.cc]: indent

File: runtime/browser/xwalk_browser_main_parts_android.cc [hunk 2]
```diff
@@ -76,6 +76,40 @@ void MoveUserDataDirIfNecessary(const base::FilePath& user_data_dir,
...
+                                    const base::FilePath& profile) {
+  if (!base::DirectoryExists(old_data_dir))
+    return;
+
+  // Local Storage
+  base::FilePath local_storage_path = old_data_dir.Append("localstorage");
+  if (base::PathExists(local_storage_path)) {
+    if (!base::Move(local_storage_path, profile.Append("Local Storage"))) {
+        NOTREACHED() << "Failed to import previous user data: localstorage";
+      }
```
Comment (d0ngjun) [inline runtime/browser/xwalk_browser_main_parts_android.cc]: indent

Comment (fujunwei) [inline runtime/browser/xwalk_browser_main_parts_android.cc]: Thanks, Done

**Comment by fujunwei** [comment]
@huningxin @lincsoon @halton @d0ngjun , PTAL.

**Comment by wuhengzhi** [comment]
In the comment, ```formate``` -> ```formats```.

**Comment by d0ngjun** [comment]
LGTM

**Comment by halton** [comment]
LGTM

**Comment by rakuco** [comment]
Is there a way for QA to test this? I'd like to have this backported only if we're sure we're not losing or damaging existing user data.

**Comment by fujunwei** [comment]
The test case is  https://github.com/crosswalk-project/crosswalk-test-suite/tree/master/usecase/usecase-wrt-android-tests/samples/ApplicationlocalStorage.  QA will test it.

**Comment by rakuco** [comment]
The test cases are only for localstorage in general, right? I was asking about testing these transitions between different Crosswalk versions and the webview.

**Pull request merged**